Rewrite first 3 paragraphs of Lua-introductions.md (#5)
IlonaSilverwood/pan-docs: Lua-introduction.md

> # Introduction

Reviewer: I know this is convention across the docs, but the "introduction" heading seems unnecessary. Something to consider in subsequent edits.

Pandoc filters allow the pandoc
abstract syntax tree (AST) to be manipulated between the parsing

> and the writing phase. [Traditional pandoc

Reviewer: ```suggestion
Pandoc filters manipulate the pandoc
abstract syntax tree (AST) between the parsing
and the writing phases. [JSON

```

Reviewer: Just trying to cut the amount of words down by avoiding passive and "causatives" like allow.

filters](https://pandoc.org/filters.html) accept a JSON
representation of the pandoc AST and produce an altered JSON
representation of the AST. They may be written in any
programming language, and invoked from pandoc using the

> `--filter` option.

Reviewer: I think this information is good, but perhaps this seems more like info that should go at the beginning of the other filters docs, or of a new page called JSON filters.

> Traditional JSON filters have limitations:

Reviewer: ```suggestion
 JSON filters have limitations:
```

Reviewer: I'm not sure whether "traditional" is such an important adjective here :smile:

1. Writing JSON to stdout and reading it from stdin (twice, once on
each side of the filter) is inefficient.

> 2. External dependencies vary between users, and universal JSON filters are not possible.

Reviewer: ```suggestion
- Writing JSON to stdout and reading it from stdin (twice, once on
each side of the filter) is inefficient.
- External dependencies vary between users, and universal JSON filters are not possible.

```

Reviewer: I'm really happy with the list here :+1: . I was thinking the same. But, since there's no procedure here, I think we can just use unordered lists.[^1]

 [^1]: https://developers.google.com/style/lists

> Pandoc 2.0 and subsequent versions make it possible to write

Reviewer: ```suggestion
Since Pandoc 2.0, you can write

```

filters in Lua without any external dependencies. A Lua
interpreter (version 5.4) and a Lua library for creating pandoc

> filters is built into the pandoc executable. Pandoc data types

Reviewer: ```suggestion
filters in Lua without any external dependencies.
The pandoc executable has a built-in Lua
interpreter (version 5.4) and a Lua library for creating pandoc
filters. Pandoc data types
```

Reviewer: I know this has nothing to do with the work in this PR, but not sure about these line breaks. I usually see either 80 characters, according to old programming standards, or with one sentence per line (my preferred way). This is 60ish per line.

Maybe in a future PR we can figure out a more comfortable way.

are marshaled to Lua directly, avoiding the overhead of writing
JSON to stdout and reading it from stdin.

> Here is an example of a Lua filter that converts strong emphasis
> to small caps:

Reviewer: ```suggestion
For example, this Lua filter converts strong emphasis
to small caps:

```

Reviewer: Every character counts! :scissors:
